client/state/posts/actions.js

```diff
@@ -62,11 +63,20 @@ export function receivePosts( posts ) {
  * @return {Function}        Action thunk
  */
 export function requestSitePosts( siteId, query = {} ) {
+	const q = {
+		...DEFAULT_POST_QUERY,
```

Comment:
I'd argue this would introduce more differences to what we have in production. There, we don't pass the following if they're not specifically set:

* `search`
* `sticky`
* `number`
* `offset`
* `page`
* `context`
* `pretty`

Note that this is not a conclusive list, it's a quick one I made while comparing to a couple of production queries.